src/core/parser/parser.js

```diff
@@ -0,0 +1,31 @@
+'use strict';
+
+import Nodes from './nodes';
+import Lexer from '../lexer/lexer';
+import { parser } from './compiled-parser';
+
+parser.lexer = new Lexer();
+parser.yy = Nodes;
+parser.yy.declarations = {};
+parser.yy.parseError = function parseError(e) {
+  throw e;
+};
+parser.yy.registerDeclaration = function registerDeclaration(
+  variableName,
+  expression,
+  explicitType,
+) {
+  if (!this.areDeclarationsHandled) {
```

Comment:
this flag is set by the runner on initial dialogue load, to handle <<declare>> commands.